nav2_costmap_2d/include/nav2_costmap_2d/observation.hpp

```diff
@@ -60,9 +60,32 @@ class Observation
   }
 
   /**
-    * @brief  Explicitly define copy assignment operator for Observation as it has a user-declared destructor
-    */
-  Observation & operator=(const Observation &) = default;
+  * @brief  Copy assignment operator
+  * @param obs The observation to copy
+  */
+  Observation & operator=(const Observation & obs){
+      origin_ = obs.origin_;
+      cloud_ = new sensor_msgs::msg::PointCloud2(*(obs.cloud_));
+      obstacle_range_ = obs.obstacle_range_;
+      raytrace_range_ = obs.raytrace_range_;
+      
+      return *this;
+  }
+
+  /**
+  * @brief  Move assignment operator
+  * @param obs The observation to move
+  */
+  Observation & operator=(Observation && obs){
```

Comment:
As well here, this is does not seem necessary

Reply:
> Added move assignment and move constructor, because it's they are called inside `std::vector` with less (de)allocations

 Just like up there. I'm not sure what exactly needed, constructor or assignment, so why not both?

Reply:
We want things to be created as needed and not have dead code in the system. If its not being used and isn't needed, then it should not be added. Its more code then we (I) need to maintain.